### PR TITLE
[Docs] Refresh tutorials

### PR DESCRIPTION
**Pull request by czechboy0**
### Motivation

Our tutorials were originally written for SOAR 0.1.0, so we need to update them to reflect the best practices as of SOAR 1.0 instead.

### Modifications

- Adapted to the new Examples structure (most importantly, point to cloning the Vapor hello world project when writing the client, instead of the GreetingService.)
- Added a mention of the shorthand APIs as an alternative to the exhaustive handling. (Still show exhaustive handling first though, as we want to make sure adopters consider that first and even prefer it.)
- More minor polish and updates.

### Result

Tutorials ready for 1.0 (sans versions, those are changed in #472)

### Test Plan

Previewed locally, actually went through all the tutorials and verified they all work.

### CODE REVIEW

File: Sources/swift-openapi-generator/Documentation.docc/Tutorials/Adding-openapi-and-swagger-ui-endpoints.tutorial [hunk 1]
```diff
@@ -44,11 +44,11 @@
         }
     }
     @Section(title: "Add a Swagger UI endpoint") {
-        Now we'll add a static `openapi.html` page that serves Swagger UI and add a redirect to this page from `/` for discoverability.
+        Now we'll add a static `openapi.html` page that serves Swagger UI and add a redirect to this page from `/openapi` for discoverability.
```
Comment (simonjbeaumont) [inline Sources/swift-openapi-generator/Documentation.docc/Tutorials/Adding-openapi-and-swagger-ui-endpoints.tutorial:47]: Either this, or the `@Section` need updating. One says the endpoint is `/openapi`, the other `openapi.yaml`.

File: Sources/swift-openapi-generator/Documentation.docc/Tutorials/_Resources/client.Package.1.swift [hunk 2]
```diff
@@ -3,12 +3,7 @@ import PackageDescription
 
 let package = Package(
     name: "GreetingServiceClient",
-    platforms: [
-        .macOS(.v10_15),
-        .iOS(.v13),
-        .tvOS(.v13),
-        .watchOS(.v6),
-    ],
+    platforms: [.macOS(.v10_15), .iOS(.v13), .tvOS(.v13), .watchOS(.v6)],
```
Comment (simonjbeaumont) [inline Sources/swift-openapi-generator/Documentation.docc/Tutorials/_Resources/client.Package.1.swift:6]: `visionOS`?

Comment (simonjbeaumont) [reply Sources/swift-openapi-generator/Documentation.docc/Tutorials/_Resources/client.Package.1.swift:6]: If we _do_ add it, let's add it to all the other places too.

File: Sources/swift-openapi-generator/Documentation.docc/Tutorials/_Resources/client.main.0.swift [hunk 3]
```diff
@@ -1 +1,4 @@
+// The Swift Programming Language
+// https://docs.swift.org/swift-book
+
```
Comment (simonjbeaumont) [inline Sources/swift-openapi-generator/Documentation.docc/Tutorials/_Resources/client.main.0.swift:3]: Do we want this?

**Review by simonjbeaumont — APPROVED** [review]

File: Sources/swift-openapi-generator/Documentation.docc/Tutorials/Adding-openapi-and-swagger-ui-endpoints.tutorial (hunk 1 above)
Comment (simonjbeaumont) [reply Sources/swift-openapi-generator/Documentation.docc/Tutorials/Adding-openapi-and-swagger-ui-endpoints.tutorial:47]: Ignore me. That's fine.

File: Sources/swift-openapi-generator/Documentation.docc/Tutorials/_Resources/client.main.0.swift (hunk 3 above)
Comment (czechboy0) [reply Sources/swift-openapi-generator/Documentation.docc/Tutorials/_Resources/client.main.0.swift:3]: Yes, it shows up in the template.

**Comment by czechboy0** [comment]
Landing this without waiting on CI, this is all in tutorials and the soundness script passed already.

**Pull request merged**